src/main/java/pl/poznan/put/cs/scenariochecker/transformations/CountSpecialStepsScenarioStrategy.java

```diff
@@ -0,0 +1,29 @@
+package pl.poznan.put.cs.scenariochecker.transformations;
+
+import pl.poznan.put.cs.scenariochecker.model.Scenario;
+import pl.poznan.put.cs.scenariochecker.model.Step;
+
+import java.util.List;
+
+public class CountSpecialStepsScenarioStrategy extends ScenarioStrategy {
+
+    @Override
+    public void processScenario(Scenario scenario) {
+        scenario.setNumberOfSpecialSteps(changeValue(scenario.getSteps()));
```

Comment:
I don't think that `changeValue` is the best method name in this situation. You should always choose the names of your methods carefully, so that they mirror the implementation.  Otherwise your code will be hard to maintain.

Reply:
Maybe `countSpecialSteps` would be better?